yt/visualization/tests/test_callbacks.py

```diff
@@ -404,6 +404,23 @@ def test_sphere_callback():
         assert_fname(p.save(prefix)[0])
 
 
+def test_invalidated_annotations():
+    # check that annotate_sphere and annotate_arrow succeed on re-running after
+    # an operation that invalidates the plot (set_font_size), see
+    # https://github.com/yt-project/yt/issues/4698
+    with _cleanup_fname() as prefix:
+        ds = fake_amr_ds(fields=("density",), units=("g/cm**3",))
+        p = SlicePlot(ds, "z", ("gas", "density"))
+        p.annotate_sphere([0.5, 0.5, 0.5], 0.1)
+        p.set_font_size(24)
+        assert_fname(p.save(prefix)[0])
```

Comment:
I think we could speed up these tests by calling `p.render` instead of `p.save` here (yes, that's also true for a lot of existing tests in this module, but of course it's way out of scope to deal with everything here)

Reply:
I'll make this change